src/fs_tree_manager/go/fs_tree_manager_grpc.pb.go

```diff
@@ -0,0 +1,314 @@
+	FSTreeManager_FetchReplica_FullMethodName  = "/fs_tree_manager.FSTreeManager/FetchReplica"
+	FSTreeManager_PullDiff_FullMethodName      = "/fs_tree_manager.FSTreeManager/PullDiff"
+	FSTreeManager_NewFSEntry_FullMethodName    = "/fs_tree_manager.FSTreeManager/NewFSEntry"
+	FSTreeManager_RemoveFSEntry_FullMethodName = "/fs_tree_manager.FSTreeManager/RemoveFSEntry"
+	FSTreeManager_PurgeReplica_FullMethodName  = "/fs_tree_manager.FSTreeManager/PurgeReplica"
+)
+
+// FSTreeManagerClient is the client API for FSTreeManager service.
+//
+// For semantics around ctx use and closing/ending streaming RPCs, please refer to https://pkg.go.dev/google.golang.org/grpc/?tab=doc#ClientConn.NewStream.
+//
+// For all RPC requests, user identifier is always needed since replicas are
+// stored separately for each user.
+//
+// We use user_id instead of user_name/user_uuid since it's more accessible:
+// - fsentry include user_id but not user_name/user_uuid
+// (https://github.com/HeyPuter/puter/blob/847b3a07a4ec59e724063f460a4c26cb62b04d42/src/backend/src/services/database/sqlite_setup/0001_create-tables.sql#L81)
+// - user_id is included in the fs events listener where user_name/user_uuid are
+// not available
+// (https://github.com/HeyPuter/puter/blob/847b3a07a4ec59e724063f460a4c26cb62b04d42/src/backend/src/services/WSPushService.js#L165-L166)
+//
+// We provide simple {New/Remove}FSEntry APIs as a straightforward way to
+// accommodate the wide variety of file system operations. These APIs should
+// always results in an coherent MerkleTree.
+type FSTreeManagerClient interface {
+	FetchReplica(ctx context.Context, in *FetchReplicaRequest, opts ...grpc.CallOption) (*MerkleTree, error)
+	PullDiff(ctx context.Context, in *PullRequest, opts ...grpc.CallOption) (*PushRequest, error)
+	// Insert a new FSEntry into the tree, update its parent's children list as
+	// well.
+	NewFSEntry(ctx context.Context, in *NewFSEntryRequest, opts ...grpc.CallOption) (*emptypb.Empty, error)
+	// Remove an FSEntry (and all its descendants) from the tree, update its
+	// parent's children list as well.
+	RemoveFSEntry(ctx context.Context, in *RemoveFSEntryRequest, opts ...grpc.CallOption) (*emptypb.Empty, error)
+	// For any fs operations that cannot be handled by New/Remove APIs, just purge
+	// the replica.
+	PurgeReplica(ctx context.Context, in *PurgeReplicaRequest, opts ...grpc.CallOption) (*emptypb.Empty, error)
+}
+
+type fSTreeManagerClient struct {
+	cc grpc.ClientConnInterface
+}
+
+func NewFSTreeManagerClient(cc grpc.ClientConnInterface) FSTreeManagerClient {
+	return &fSTreeManagerClient{cc}
+}
+
+func (c *fSTreeManagerClient) FetchReplica(ctx context.Context, in *FetchReplicaRequest, opts ...grpc.CallOption) (*MerkleTree, error) {
+	cOpts := append([]grpc.CallOption{grpc.StaticMethod()}, opts...)
+	out := new(MerkleTree)
+	err := c.cc.Invoke(ctx, FSTreeManager_FetchReplica_FullMethodName, in, out, cOpts...)
+	if err != nil {
+		return nil, err
+	}
+	return out, nil
```

Comment:
I'm nitpicking here but why not just `return out, nil` since `out` is already a nil-able `*MerkelTree` and `err` will have the appropriate value?
```suggestion
	return out, err
```

Reply:
`src/fs_tree_manager/go` and `src/fs_tree_manager/js` are generated by protobuf, no need to review